Classes/Aspects/ThumbnailAspect.php

```diff
@@ -140,4 +130,21 @@ protected function getRequest(JoinPointInterface $joinPoint): ActionRequest
     {
         return $joinPoint->getMethodArgument('request');
     }
+
+    public function getSourceUri(Asset $asset): string
```

Comment:
Since we only ever access the resource of the asset, the method would be better reusable if we accept just a `PersistentResource`.